pennylane/ops/projection.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the available built-in quantum projection operations
+supported by PennyLane.
+"""
+
+import numpy as np
+
+from pennylane.operation import AnyWires, Projection
+
+
+class Measure(Projection):
+    r"""Measure(wires)
+    A measurement in the computational basis.
+
+    **Details:**
+
+    * Number of wires: Any
+    * Number of parameters: 0
+    * Gradient recipe: None
+
+    Args:
+        wires (Sequence[int] or int): the wire(s) the operation acts on
+
+    .. UsageDetails::
+
+        The `Measurement` operation is applied to the wire we
+        want to measure. The return of a circuit is then evaluated on
+        the output state conditioned on the given measurement outcome.
+        This will depend on the random outcome of the measurement,
+        so expectation values should be obtained over several shots.
+
+        For instance, consider the following circuit:
+
+        .. code-block:: python3
+
+           dev = qml.device('default.qubit', wires=2)
+
+            @qml.qnode(dev)
+            def circuit():
+                qml.Hadamard(wires=0)
+                qml.CNOT(wires=[0, 1])
+                qml.Measure(wires=[0, 1])
+
+                return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+
+        By running this circuit, we would get an output of the form:
+
+        >>> circuit()
+        array([1., 1.])
+
+        which corresponding to a measurement of :math:`|00\rangle`.
```

Comment:
Do we want to add that if we were to run again, the result might be different. (e.g., sometimes get [-1, -1])